python/lsst/ip/isr/crosstalk.py

```diff
@@ -23,17 +23,441 @@
 Apply intra-detector crosstalk corrections
 """
 import numpy as np
+from astropy.table import Table
 
 import lsst.afw.math
-import lsst.afw.table
 import lsst.afw.detection
 from lsst.pex.config import Config, Field, ChoiceField, ListField
 from lsst.pipe.base import Task
 
-__all__ = ["CrosstalkConfig", "CrosstalkTask", "subtractCrosstalk", "writeCrosstalkCoeffs",
+from lsst.ip.isr import IsrCalib
+
+
+__all__ = ["CrosstalkCalib", "CrosstalkConfig", "CrosstalkTask",
            "NullCrosstalkTask"]
 
 
+class CrosstalkCalib(IsrCalib):
+    """Calibration of amp-to-amp crosstalk coefficients.
+
+    Parameters
+    ----------
+    detector : `lsst.afw.cameraGeom.Detector`, optional
+        Detector to use to pull coefficients from.
+    log : `lsst.log.Log`, optional
+        Log to write messages to.
+    **kwargs :
+        Parameters to pass to parent constructor.
+    """
+    _OBSTYPE = 'CROSSTALK'
+    _SCHEMA = 'Gen3 Crosstalk'
+    _VERSION = 1.0
+
+    def __init__(self, detector=None, **kwargs):
+        self.hasCrosstalk = False
+        self.nAmp = 0
+        self.crosstalkShape = (self.nAmp, self.nAmp)
+
+        self.coeffs = None
+        self.coeffErr = None
+        self.coeffNum = None
+        self.interChip = None
+
+        super().__init__(**kwargs)
+        self.requiredAttributes.update(['hasCrosstalk', 'nAmp', 'coeffs',
+                                        'coeffErr', 'coeffNum', 'interChip'])
+        if detector:
+            self.fromDetector(detector)
+
+    def updateMetadata(self, setDate=False, **kwargs):
+        """Update calibration metadata.
+
+        This calls the base class's method after ensuring the required
+        calibration keywords will be saved.
+
+        Parameters
+        ----------
+        setDate : `bool`, optional
+            Update the CALIBDATE fields in the metadata to the current
+            time. Defaults to False.
+        kwargs :
+            Other keyword parameters to set in the metadata.
```

Comment:
Should this be type `dict`? I've never seen kwargs documented quite like this.

Reply:
Updated with `dict` and `collections.abc.Mappings`, based on the upstream version: https://github.com/lsst/daf_base/blob/master/python/lsst/daf/base/propertyContainer/propertyContainerContinued.py#L469.